### PR TITLE
emailTemplates API

### PR DESCRIPTION
**Pull request by Firuz2006**

### CODE REVIEW

File: src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using MRA.Identity.Application.Contract.EmailTemplates.Commands;
+using MRA.Identity.Application.Contract.EmailTemplates.Queries;
+
+namespace MRA.Identity.Api.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+[Authorize]
```
Comment (komdil) [inline src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:11]: put an Administrator role

File: src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    : ControllerBase
+{
+    [HttpPost]
+    public async Task<IActionResult> Create([FromBody] CreateEmailTemplateCommand createCommand)
+    {
+        var slug = await mediator.Send(createCommand);
+        return Ok(slug);
+    }
+
+    [HttpDelete]
```
Comment (komdil) [inline src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:22]: [HttpDelete("{slug}")]

File: src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    [HttpPut]
+    public async Task<IActionResult> Update([FromBody] UpdateEmailTemplateCommand updateCommand)
+    {
+        var slug = await mediator.Send(updateCommand);
+        return Ok(slug);
+    }
+
+    [HttpGet("getTemplate")]
```
Comment (komdil) [inline src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:36]: [HttpDelete("{slug}")]

File: src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public async Task<IActionResult> GetTemplate([FromQuery] string slug)
+    {
+        var emailTemplateResponse = await mediator.Send(new GetEmailTemplateQuery { Slug = slug });
+        return Ok(emailTemplateResponse);
+    }
+
+    [HttpGet("getSubjects")]
```
Comment (komdil) [inline src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:43]: Do we need this?

File: src/MRA.Identity.Application.Contract/EmailTemplates/Commands/UpdateEmailTemplateCommand.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using MediatR;
+
+namespace MRA.Identity.Application.Contract.EmailTemplates.Commands;
+#nullable enable
```
Comment (komdil) [inline src/MRA.Identity.Application.Contract/EmailTemplates/Commands/UpdateEmailTemplateCommand.cs:4]: remove it. Put required or set the default value instead

**Review by komdil — CHANGES_REQUESTED** [review]

File: src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs (hunk 4 above)
Comment (Firuz2006) [reply src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:43]: yes because controller has two get methods

Comment (komdil) [reply src/MRA.Identity.Api/Controllers/EmailTemplatesController.cs:43]: Change [HttpGet("getSubjects")] to [HttpGet]. Route should be /emailtempates

**Review by komdil — CHANGES_REQUESTED** [review]

**Review by komdil — APPROVED** [review]

**Pull request merged**